PayoutCurve.md

```diff
@@ -125,36 +186,74 @@ Please note that while the following may seem complex, it should boil down to ve
-  can be cached for each `i` in a polynomial piece, call this `coef_i`.
-  For a given `event_outcome` let `all_prod = PROD(i = 0, i < points.length, event_outcome - points(i).event_outcome)`.
+1. type: ??? (`hyperbola_payout_curve_piece`)
+2. data:
+   * [`bool`:`use_positive_piece`]
+   * [`bool`:`translate_outcome_sign`]
+   * [`bigsize`:`translate_outcome`]
+   * [`u16`:`translate_outcome_extra_precision`]
+   * [`bool`:`translate_payout_sign`]
+   * [`bigsize`:`translate_payout`]
+   * [`u16`:`translate_payout_extra_precision`]
+   * [`bool`:`a_sign`]
+   * [`bigsize`:`a`]
+   * [`u16`:`a_extra_precision`]
+   * [`bool`:`b_sign`]
+   * [`bigsize`:`b`]
+   * [`u16`:`b_extra_precision`]
+   * [`bool`:`c_sign`]
+   * [`bigsize`:`c`]
+   * [`u16`:`c_extra_precision`]
+   * [`bool`:`d_sign`]
+   * [`bigsize`:`d`]
+   * [`u16`:`d_extra_precision`]
+
+If `use_positive_piece` is set to true, then `y_1` is used, otherwise `y_2` is used.
+
+Then there are six numeric values represented as a `bool` sign, a `bigsize` integer, and a `u16` extra precision.
+To be precise, the numbers used should be: (`num_sign`)( `num + double(num_extra_precision) >> 16`).
 
-  The sum can then be computed as `SUM(i = 0, i < points.length, coef_i * all_prod / (event_outcome - points(i).event_outcome))`.
+Note that this `payout_function` is from the offerer's point of view.
+To evaluate the accepter's `payout_function`, you must evaluate the offerer's `payout_function` at a given
+`event_outcome` and subtract the resulting payout from `total_collateral`.
 
-* When precision ranges are introduced, derivatives of the polynomial can be used to reduce the number of calculations needed.
-  For example, when dealing with a cubic piece, if you are going left to right and enter a new value modulo precision while the first derivative (slope) is positive and the second derivative (concavity) is negative then you can take the tangent line to the curve at this point and find it's intersection with the next value boundary modulo precision.
-  If the derivatives' signs are the same, then the interval from the current x-coordinate to the x-coordinate of the intersection is guaranteed to all be the same value modulo precision.
+#### Requirements
+
+* `a*d` MUST be not equal `b*c`.
+* The resulting curve MUST be defined for every `event_outcome` (without division by zero).
+
+#### Hyperbola Evaluation
+
+The two curve pieces from which one is chosen by a boolean flag, are just the parameterization of the `y-coordinate`
+in the above expression by `x` when the expression is expanded:
+
+```
+y_1 = c * (x - f_1 + sqrt((x - f_1)^2 - 4*a*b))/(2*a) + 2*a*d/(x - f_1 + sqrt((x - f_1)^2 - 4*a*b)) + f_2
+
+y_2 = c * (x - f_1 - sqrt((x - f_1)^2 - 4*a*b))/(2*a) + 2*a*d/(x - f_1 - sqrt((x - f_1)^2 - 4*a*b)) + f_2
+```
+
+We will refer to `y_1` as the positive piece and `y_2` as the negative piece, only because they use positive
+and negative square roots respectively.
```

Comment:
Might be good to add a section for Optimized Evaluation for getting the outcome for a payout.

Using the inverse of `y_1` or `y_2` you can easily calculate the outcome without having to calculate all possible payouts using `y_1` or `y_2`.

https://github.com/atomicfinance/node-dlc/blob/ca9f275a375ca3e3ae527f5e6f14ef4e83bf76db/packages/core/lib/dlc/HyperbolaPayoutCurve.ts#L53